feat: include a short summary of the blog post.

robbielyman opened the pull request:
Since the post's `content` field is already populated with the contents of a `summary` tag (as far as I can see), this commit hooks that data into the message posted to Zulip.

i'd love guidance on how to test my changes! i was able to build the docker image, but running it appears to just drop me into a Python REPL.

- [x] Test
- [x] Screenshot
- [x] CHANGES.md entry
- [x] Documentation

robbielyman committed:
feat: include a short summary of the blog post.

Since the post's `content` field is already populated with the contents
of a `summary` tag (as far as I can see), this commit hooks that data
into the message posted to Zulip.

sursh commented on home/zulip_helpers.py:
````diff
@@ -17,7 +18,8 @@
 MEMBERS_URL = "https://recurse.zulipchat.com/api/v1/users"
-ANNOUNCE_MESSAGE = "{} has a new blog post: [{}]({})"
+ANNOUNCE_MESSAGE = "{} has a new blog post: [{}]({})\n```quote\n{}\n```"
+PLACEHOLDER = "Consider adding a `summary` to your next blog post! It'll show up here."
````
Suggestion: assuming this copy would go out on each post to Zulip, rather than putting them on blast, consider not displaying a summary if one's missing. 

Either way, consider adding clear instructions so people can learn how to get blaggregator to find the summary on their feed if they want—either in the Zulip message (if you're putting them on blast) or the README (in both cases)

robbielyman replied:
great feedback! I'll make that change.

robbielyman committed:
fix: don't put folks on blast for missing summaries.

also, remove the dependency on `textwrap`.

adds a usage description of the feature to the README
and an entry to CHANGES.md

robbielyman commented:
just wanted to re-up the ask for suggestions on how to test this change :)

sursh commented:
Nice! 

Re: testing, have you checked out the directions linked from the README yet? I don’t know if they’re up to date, but seems like a good place to start

robbielyman commented:
i'm a little stuck on the dev directions (notably Docker and Postgres are both new to me, so probably the error is small but tough for me to see). i've installed `Postgres.app` and been able to create a database and a user with `CREATEDB` powers, but when i run `docker-compose up` the `db` process fails to start with an error saying the "postgres" user's password authentication failed.

robbielyman committed:
feat: markdownify summaries

Use markdownify in feedergrabber27.py to convert HTML summaries to
markdown ones.

robbielyman commented:
I was able to test this, thanks to @jdkaplan :). I think he'll PR the dev environment fix that got me unblocked.

There's a problem: `post.content` is formatted as HTML, but Zulip sanitizes HTML content and renders only Markdown. I'm not sure if this is easily solvable without literally decompiling the HTML into Markdown, which seems a bit out of scope.

For example, I added [Maddie's blog](https://ntietz.com) and this is the message that my local version of Blaggregator produced:

> Sending message "**Robbie Lyman** has a new blog post: [Covers as a way of learning music and code](http://localhost:8000/post/kpKMc6/view)
> ```quote
> <p>When you're just getting started with music, you have so many skills to learn.
> You have to be able to play your instrument and express yourself through it.
> You need to know the style you're playing, and its idioms and conventions.
> You may want to record your music, and need all the skills that come along with it.</p>
> <p>Music is, mostly, subjective: there's not an objective right or wrong way to do things.
> And that can make it really hard!
> Each of these skills is then c...
> ```
> " to bot-test (stream)

sursh commented:
idea: maybe an easier place to intervene is at crawl time? see `feedergrabber27.py`. I'm thinking that we're already parsing HTML there, so it may be an easier lift to just parse it a little differently. 

Since this change would only affect new entries (and I don't _think_ that post.content is used anywhere else at all, but you should fact-check that) then there's no need to handle retroactive entries, only new ones. 

(this idea is based on 5 min of flipping through the codebase and my dusty memories of my initial work on the project, but I haven't been closely following changes since, nor have I set it up to run locally, so please vet the idea before using it!)

robbielyman commented:
thanks for the suggestion @sursh! I think that got me in the right direction; here's a new test with Maddie's blog:

> **Robbie Lyman** has a new blog post: [Covers as a way of learning music and code](http://localhost:8000/post/gdDxso/view)
> ```quote
> When you're just getting started with music, you have so many skills to learn.
> You have to be able to play your instrument and express yourself through it.
> You need to know the style you're playing, and its idioms and conventions.
> You may want to record your music, and need all the skills that come along with it.
> 
> Music is, mostly, subjective: there's not an objective right or wrong way to do things.
> And that can make it really hard!
> Each of these skills is then couched in...
> ```

and

> **Robbie Lyman** has a new blog post: [That boolean should probably be something else](http://localhost:8000/post/feiCVZ/view)
> ```quote
> One of the first types we learn about is the boolean.
> It's pretty natural to use, because boolean logic underpins much of modern computing.
> And yet, it's one of the types we should probably be using a lot less of.
> In almost every single instance when you use a boolean, it should be something else.
> 
> The trick is figuring out *what* "something else" is.
> Doing this is worth the effort.
> It tells you a lot about your system, and it will improve your design (even if you end up u...
> ```

both of these seem much better equipped to be Zulip messages :)
